lib/SILGen/SILGenExpr.cpp

```diff
@@ -3521,6 +3521,14 @@ SILGenModule::emitKeyPathComponentForDecl(SILLocation loc,
     return storage->isSettable(storage->getDeclContext());
   };
 
+  // We cannot use the same opened archetype in the getter and setter. Therefore
+  // we create a new one for the setter.
```

Comment:
Doesn't the getter also share an opened archetype with the original expression?

Reply:
What I've seen is that the opened archetype is only used in the getter and the setter. So making a new one for the setter is sufficient

Reply:
On the other hand, it does not harm to create new opened archetypes for the getter as well